internal/initwd/module_install.go

```diff
@@ -49,6 +49,22 @@ type moduleVersion struct {
 	version string
 }
 
+type TypeDiagnosticExtra string
+
+const TypeModuleVersionDeprecation = "module_version_deprecation"
+
+type ModuleVersionDeprecationDiagnosticExtra struct {
```

Comment:
Once this is merged, fields can't be modified or removed. Can you add some godoc comments about this being public API?

Reply:
Include 👍  https://github.com/hashicorp/terraform/pull/35231/commits/838f23b5f94dccbbcd8fc9a8dedacc294b72944c